src/createContext.spec.lua

```diff
@@ -0,0 +1,168 @@
+				return nil
+			end
+
+			local element = createElement(context.Provider, {
+				value = "Test",
+			}, {
+				Listener = createElement(Listener),
+				Listener2 = createElement(Listener),
+			})
+
+			expect(function()
+				local tree = noopReconciler.mountVirtualTree(element, nil, "Provide Tree")
+				noopReconciler.unmountVirtualTree(tree)
+			end).to.throw()
+		end)
+	end)
+
+	describe("Consumer", function()
+		it("should expect a render function", function()
+			local context = createContext("Test")
+
+			local Listener = PureComponent:extend("Listener")
+
+			function Listener:render()
+				return nil
+			end
+
+			local element = createElement(context.Consumer)
+
+			expect(function()
+				local tree = noopReconciler.mountVirtualTree(element, nil, "Provide Tree")
+				noopReconciler.unmountVirtualTree(tree)
+			end).to.throw()
+		end)
+
+		it("should return the default value if there is no Provider", function()
+			local foundValue
+			local context = createContext("Test")
+
+			local element = createElement(context.Consumer, {
+				render = function(value)
+					foundValue = value
+				end,
+			})
+
+			local tree = noopReconciler.mountVirtualTree(element, nil, "Provide Tree")
+			noopReconciler.unmountVirtualTree(tree)
+
+			expect(foundValue).to.equal("Test")
+		end)
+
+		it("should pass the value to the render function", function()
+			local foundValue
+			local context = createContext("Test")
+
+			local Listener = PureComponent:extend("Listener")
+
+			function Listener:render()
+				return createElement(context.Consumer, {
+					render = function(value)
+						foundValue = value
+					end,
+				})
+			end
+
+			local element = createElement(context.Provider, {
+				value = "NewTest",
+			}, {
+				Listener = createElement(Listener),
+			})
+
+			local tree = noopReconciler.mountVirtualTree(element, nil, "Provide Tree")
+			noopReconciler.unmountVirtualTree(tree)
+
+			expect(foundValue).to.equal("NewTest")
+		end)
+
+		it("should update when the value updates", function()
+			local renderCount = 0
+			local foundValue
+
+			local context = createContext("Test")
+
+			local Listener = PureComponent:extend("Listener")
+
+			function Listener:render()
+				return createElement(context.Consumer, {
+					render = function(value)
+						renderCount = renderCount + 1
+						foundValue = value
+					end,
+				})
+			end
+
+			local element = createElement(context.Provider, {
+				value = "NewTest",
+			}, {
+				Listener = createElement(Listener),
+			})
+
+			local tree = noopReconciler.mountVirtualTree(element, nil, "Provide Tree")
+
+			expect(renderCount).to.equal(1)
+			expect(foundValue).to.equal("NewTest")
+
+			noopReconciler.updateVirtualTree(tree, createElement(context.Provider, {
+				value = "ThirdTest",
+			}, {
+				Listener = createElement(Listener),
+			}))
+
+			expect(renderCount).to.equal(2)
+			expect(foundValue).to.equal("ThirdTest")
+
+			noopReconciler.unmountVirtualTree(tree)
+		end)
```

Comment:
This test can also be rewritten in terms of Roact's spies, which might be more readable.